doc/luasnip.txt

```diff
@@ -3628,197 +3673,423 @@ GENERAL ~
+
+
+filetype_extend(ft, extend_ft)                   *luasnip-api-filetype_extend*
+
+Add `extend_ft` filetype to inherit its snippets from `ft`. Example:
+
+>lua
+    ls.filetype_extend("sh", {"zsh"})
+    ls.filetype_extend("sh", {"bash"})
+<
+
+This makes all `sh` snippets available in `sh`/`zsh`/`bash` `zsh`/`bash` `bash`
+buffers.
+
+- `ft: string`
+- `extend_ft: string[]`
+
+
+filetype_set(ft, fts)                               *luasnip-api-filetype_set*
+
+Set `fts` filetypes as inheriting their snippets from `ft`. Example:
+
+>lua
+    ls.filetype_set("sh", {"sh", "zsh", "bash"})
+<
+
+This makes all `sh` snippets available in `sh`/`zsh`/`bash` `zsh`/`bash` `bash`
+buffers.
+
+- `ft: string`
+- `fts: string[]`
+
+
+cleanup()                                                *luasnip-api-cleanup*
+
+Clear all loaded snippets. Also sends the `User LuasnipCleanup` autocommand, so
+plugins that depend on luasnip’s snippet-state can clean up their
+now-outdated state.
+
+
+refresh_notify(ft)                                *luasnip-api-refresh_notify*
+
+Trigger the `User LuasnipSnippetsAdded` autocommand that signifies to other
+plugins that a filetype has received new snippets.
+
+- `ft: string` The filetype that has new snippets. Code that listens to this event can retrieve this
+    filetype from `require("luasnip").session.latest_load_ft`.
+
+
+setup_snip_env()                                  *luasnip-api-setup_snip_env*
+
+Injects the fields defined in `snip_env`, in `setup`, into the callers global
+environment. This means that variables like `s`, `sn`, `i`, `t`, … (by
+default) work, and are useful for quickly testing snippets in a buffer:
+
+>lua
+    local ls = require("luasnip")
+    ls.setup_snip_env()
+    
+    ls.add_snippets("all", {
+        s("choicetest", {
+            t":", c(1, {
+                t("asdf", {node_ext_opts = {active = { virt_text = {{"asdf", "Comment"}} }}}),
+                t("qwer", {node_ext_opts = {active = { virt_text = {{"qwer", "Comment"}} }}}),
+            })
+        })
+    }, { key = "3d9cd211-c8df-4270-915e-bf48a0be8a79" })
+<
+
+where the `key` makes it easy to reload the snippets on changes, since the
+previously registered snippets will be replaced when the buffer is re-sourced.
+
+
+get_snip_env(): table                               *luasnip-api-get_snip_env*
+
+Return the currently active snip_env.
+
+
+get_id_snippet(id): LuaSnip.Snippet               *luasnip-api-get_id_snippet*
+
+Get the snippet corresponding to some id.
+
+- `id: LuaSnip.SnippetID`
+
+
+add_snippets(ft?, snippets, opts?)                  *luasnip-api-add_snippets*
+
+Add snippets to luasnip’s snippet-collection. NOTE: Calls `refresh_notify` as
+needed if enabled via `opts.refresh_notify`.
+
+- `ft?: string?` The filetype to add the snippets to, or nil if the filetype is specified in
+    `snippets`.
+- `snippets: (LuaSnip.Addable[]|{ [string]: LuaSnip.Addable[] })` If `ft` is nil a table mapping a
```

Comment:
note: I think we're missing some docs about what `LuaSnip.Addable` means, can be 'added' (👀) later though.

Reply:
Yeah yeah, see what you did there :D
I'll add a small explanation for now (in short, `LuaSnip.Addable` is a container for one or more expandable snippets, it was introduced to support `multiSnippet` where adding one object could require adding more than one snippet)